File: Maximum Depth of Binary Tree/Maximum Depth of Binary Tree.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param root, a tree node
    # @return an integer
    def maxDepth(self, root):
        if(root == None):
            return 0
        left = self.maxDepth(root.left) + 1
        right = self.maxDepth(root.right) + 1
        return max(left, right)

File: Maximum Depth of Binary Tree/Maximum Depth of Binary Tree.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:

    int recur(TreeNode* node){
        if(!node)   return 0;
        int left = recur(node->left);
        int right = recur(node->right);
        return (left > right) ? left + 1 : right + 1;
    }

    int maxDepth(TreeNode *root) {
        return recur(root);
    }
};
